tests/aggregation/tests.py

```diff
@@ -2158,6 +2159,79 @@ def test_having_with_no_group_by(self):
         )
         self.assertEqual(list(author_qs), [337])
 
+    def test_JSONArrayAgg(self):
+        vals = Store.objects.aggregate(jsonarrayagg=JSONArrayAgg("name"))
+        self.assertEqual(
+            vals,
+            {"jsonarrayagg": ["Amazon.com", "Books.com", "Mamma and Pappa's Books"]},
+        )
+
+    def test_JSONArrayAgg_datefield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("pubdate"))
+        self.assertEqual(
+            vals,
+            {
+                "jsonarrayagg": [
+                    "2007-12-06",
+                    "2008-03-03",
+                    "2008-06-23",
+                    "2008-11-03",
+                    "1995-01-15",
+                    "1991-10-15",
+                ]
+            },
+        )
+
+    def test_JSONArrayAgg_decimalfield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("price"))
+        self.assertEqual(
+            vals, {"jsonarrayagg": [30.0, 23.09, 29.69, 29.69, 82.8, 75.0]}
+        )
+
+    def test_JSONArrayAgg_integerfield(self):
+        vals = Book.objects.aggregate(jsonarrayagg=JSONArrayAgg("pages"))
+        self.assertEqual(vals, {"jsonarrayagg": [447, 528, 300, 350, 1132, 946]})
+
+    def test_JSONArrayAgg_filter(self):
+        vals = Author.objects.aggregate(
+            jsonarrayagg=JSONArrayAgg("age", filter=Q(age__gt=29))
+        )
+        if connection.vendor == "mysql":
+            self.assertEqual(
+                vals, {"jsonarrayagg": [34, 35, 45, None, 37, None, None, 57, 46]}
```

Comment:
This is artifact of MySQl not supporting the `FILTER` clause for aggregate function and requiring its emulation with `CASE` and `WHEN`. This one will be hard to solve as we'll likely need to express the aggregation as a subquery.